lib/sparkql/evaluator.rb

```diff
@@ -0,0 +1,152 @@
+# Using an external source for resolving the individual sparkql expressions,
+# this class will evaluate the rest of a parsed sparkql string to true or false.
+# Namely, this class will handle all the nesting, boolean algebra, and dropped 
+# fields. Plus, it has some optimizations built in to skip the processing for
+# any expressions that don't contribute to the net result of the filter.
```

Comment:
> Using an external source for resolving...

Maybe note what we're expecting `expression_resolver` to accept messages for.  It took me a bit (granted, today is a Friday for me so I may be a bit slow :-)) to realize "external source for resolving the individual expressions" meant something that'll `resolve()` each `expression` and return true or false if the data the SparkQL is being compared against matches.